currency-rates/internal/scheduler/subscription_cron.go

```diff
@@ -0,0 +1,40 @@
+package scheduler
+
+import (
+	"context"
+
+	"github.com/RinaDish/currency-rates/internal/services"
+	"github.com/go-co-op/gocron/v2"
+	"github.com/RinaDish/currency-rates/tools"
+)
+
+type Cron struct {
+	logger  logger.Logger
+	cron  gocron.Scheduler
+}
+
+func NewCron(logger logger.Logger, ctx context.Context, subscriptionService services.SubscriptionService) Cron {
```

Comment:
Maybe better to separate Cron initialization from adding new handlers to cron?
LIke Add method RegisterTask which will take interfaces due to SOLID principles?

Reply:
Done